ranger/core/runner.py

```diff
@@ -226,13 +226,26 @@ def __call__(
                 wait_for_enter = True
         if 'r' in context.flags:
             # TODO: make 'r' flag work with pipes
-            if 'sudo' not in get_executables():
-                return self._log("Can not run with 'r' flag, sudo is not installed!")
+            # XXX: is there any way to make it better?
+            root_commands = ['sudo', 'doas', 'run0']
+            root_command = None
+            for tested_command in root_commands:
+                if tested_command not in get_executables():
+                    continue
```

Comment:
```python
for x in seq:
   if not condition:
      continue
   else:
      # do smth
```
Could it be simplified, avoiding one of `if` branches and `continue`?

```python
for x in seq:
   if condition:
      # do smth
```
---
BTW, an alternative way to reach the same result in a single assignment, instead of nested `for` + `if`:

```python
root_command = next(
   (cmd in root_commands if cmd in get_executables()),
   None,
)
```
generator expression is used to filter out elements, and
built-in [`next()`](https://docs.python.org/2.7/library/functions.html#next) is used to select first element of sequence, or `None` if empty